processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -232,23 +249,45 @@ public CursorSequenceBuilder(
 
       final GenericColumn timestamps = index.getColumn(Column.TIME_COLUMN_NAME).getGenericColumn();
 
+      Iterable<Long> iterable = gran.iterable(interval.getStartMillis(), interval.getEndMillis());
+      if (descending) {
+        iterable = Lists.reverse(ImmutableList.copyOf(iterable));
+      }
+
+      // to remove branch in a loop
+      final Predicate<long[]> breakCondition = descending ? new Predicate<long[]>()
```

Comment:
Technically, just doing this predicate won't remove the branch, it will move it into the dynamic function call for the `Predicate` interface.  If you want to remove the branch, you have to actually do the branch once before you call into the loop (i.e. build an object that does the dynamic function call before the loop).